paddle/fluid/operators/batch_norm_op.cu.cc

```diff
@@ -118,15 +122,16 @@ class BatchNormKernel<platform::CUDADeviceContext, T>
 
     // alloc memory
     y->mutable_data<T>(ctx.GetPlace());
```

Comment:
why here does not use BatchNormParamType?

Reply:
Per the cudnn API, when we are using fp16 mode to run Batch norm cudnn, the data type of input `x` and output `y` will be fp16, but the other input parameters including `mean` and `variance` will still be float32 type.

So here `T == float16`, and `BatchNormParamType<T> == float` as defined in `cudnn_helper.h`